src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs

```diff
@@ -364,196 +359,129 @@ private int RunCore(TextWriter consoleOutput, CancellationToken cancellationToke
 
             EmitResult emitResult;
 
-            // EDMAURER: Don't yet know if there are method body errors. don't overwrite
-            // any existing output files until the compilation is known to be successful.
-            string tempExeFilename = null;
-            string tempPdbFilename = null;
-
             // NOTE: as native compiler does, we generate the documentation file
             // NOTE: 'in place', replacing the contents of the file if it exists
 
-            try
-            {
-                FileStream output = CreateTempFile(consoleOutput, out tempExeFilename);
+            string finalOutputPath;
+            string finalPdbFilePath;
+            string finalXmlFilePath;
 
-                // Can happen when temp directory is "full"
-                if (output == null)
-                {
-                    return Failed;
-                }
+            FileStream xml = null;
 
-                string finalOutputPath;
-                string finalPdbFilePath;
-                string finalXmlFilePath;
+            cancellationToken.ThrowIfCancellationRequested();
 
-                using (output)
+            finalXmlFilePath = Arguments.DocumentationPath;
+            if (finalXmlFilePath != null)
+            {
+                xml = OpenFile(finalXmlFilePath, consoleOutput, FileMode.OpenOrCreate, FileAccess.Write, FileShare.ReadWrite | FileShare.Delete);
```

Comment:
I'm not opposed to changing the behavior, I just wanted to clarify why I didn't make a change here.